app/src/main/java/org/p2p/wallet/striga/model/StrigaDataLayerError.kt

```diff
@@ -9,13 +9,15 @@ sealed class StrigaDataLayerError(override val message: String) : Throwable() {
         body: String = cause.errorBodyOrNull().orEmpty()
     ) : StrigaDataLayerError("Striga API unavailable, code: ${cause.code()} body: $body")
 
-    class ApiServiceError(
+    open class ApiServiceError(
         val response: StrigaApiErrorResponse,
     ) : StrigaDataLayerError("Striga API returned error: code=${response.errorCode.code} details=${response.details}") {
         val errorCode: StrigaApiErrorCode
             get() = response.errorCode
     }
 
+    class PhoneNumberAlreadyUsedError(response: StrigaApiErrorResponse) : ApiServiceError(response)
```

Comment:
we have StrigaApiErrorCode for this kind of things to remove inheritance
it doesn't work?

Reply:
Specializing errors helps to avoid duplicating code when extracting error codes. If you're not in favor of using inheritance in this case, I can create a separate error class with the same constructor instead.